requirements.txt

```diff
@@ -13,10 +13,18 @@ pytest
 pytest-cov
 raven
 requests
+requirements-parser
 scipy
 scikit-image
 scikit-learn
 tensorflow==1.5.0
+# PyTorch's Linux/Windows distribution is very large due to its GPU support,
+# but we only need that for training models. For users, use the CPU-only version
+# (only available directly from the PyTorch project).
+# The OS X version has never had GPU support, so doesn't need the workaround.
+-f https://download.pytorch.org/whl/cpu/torch_stable.html
+torch==1.5.0+cpu; sys_platform != "darwin"
+torch==1.5.0; sys_platform == "darwin"
```

Comment:
Is it possible to make this conditional? we may want to install to get the gpu version by default when we're not on travis

Reply:
It's a bit hard to depend on Pytorch having CUDA without using a system package manager like Nix since conda/pip don't know what cuda version your system is running (see https://pytorch.org/get-started/locally/) and trying to guess/autodetect also seems like a mess.

Reply:
@cfhammill inference time is so fast on modern CPUs (a few seconds for an entire volume) that i don't think we need to bother maintaining GPU version of Pytorch.

Reply:
Oops github-email didn't put my comment in the right thread: I think we can handle different use cases nicely if we write up setup.py and use `extras_requires`: https://github.com/neuropoly/spinalcordtoolbox/pull/2715#issuecomment-629878087. Another approach I've seen is to maintain two requirements.txts: the main one, and a requirements-dev.txt, but I think `extras` is nicer.